bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/graphics/Image.java

```diff
@@ -1110,11 +1160,33 @@ public ImageData getImageData (int zoom) {
 	} else if (imageFileNameProvider != null) {
 		ElementAtZoom<String> fileName = DPIUtil.validateAndGetImagePathAtZoom (imageFileNameProvider, zoom);
 		return DPIUtil.scaleImageData (device, new ImageData (fileName.element()), zoom, fileName.zoom());
+	} else if (imageGcDrawer != null) {
+		return drawWithImageGcDrawer(width, height, zoom);
 	} else {
 		return DPIUtil.scaleImageData (device, getImageDataAtCurrentZoom (), zoom, currentDeviceZoom);
 	}
 }
 
+
+
+private ImageData drawWithImageGcDrawer(int width, int height, int zoom) {
+	if (this.imageGcDrawer != null) {
+		Image image = new Image(device, width, height);
+		GC gc = new GC(image);
+		try {
+			imageGcDrawer.drawOn(gc);
+			ImageData imageData = image.getImageData(zoom);
+			imageGcDrawer.postProcess(imageData);
+			return imageData;
+		} finally {
+			gc.dispose();
+			image.dispose();
+		}
+	}
+	SWT.error(SWT.ERROR_INVALID_ARGUMENT, null, ": ImageGcDrawer [" + imageGcDrawer + "] is null.");
```

Comment:
I do not see any comparable way of throwing an error because of a null provider in the GTK implementation of Image and, if I don't miss anything, this line of code can never be reached anyway, as all callers properly ensure that the imageGcDrawer is null. So I would propose to complete remove the if block and have the drawer being null a precondition for caling that method.